acquire-driver-common/src/simcams/simulated.camera.c

```diff
@@ -239,35 +259,43 @@ simulated_camera_streamer_thread(struct SimulatedCamera* self)
                   "Unexpected index for the kind of simulated camera. Got: %d",
                   self->kind);
         }
-        {
+
+        // apply binning if applicable
+        if (self->properties.binning > 1) {
             int w = full.dims.width;
             int h = full.dims.height;
             int b = self->properties.binning >> 1;
             while (b) {
-                ECHO(bin2(self->im.data, w, h));
+                ECHO(bin2(self->im.render_data, w, h));
                 b >>= 1;
                 w >>= 1;
                 h >>= 1;
             }
         }
 
-        if (self->properties.input_triggers.frame_start.enable) {
-            while (!self->software_trigger.triggered) {
-                ECHO(condition_variable_wait(
-                  &self->software_trigger.trigger_ready, &self->im.lock));
-            }
-            self->software_trigger.triggered = 0;
+        ++frame_id;
+
+        // sleep for the remainder of the exposure time
+        float toc = (float)clock_toc_ms(&self->streamer.throttle);
+        if (self->streamer.is_running) {
+            clock_sleep_ms(&self->streamer.throttle, exposure_time_ms - toc);
         }
 
-        self->hardware_timestamp = clock_tic(0);
-        ++self->im.frame_id;
+        if (self->im.frame_wanted) {
+            ECHO(lock_acquire(&self->im.lock));
 
-        ECHO(condition_variable_notify_all(&self->im.frame_ready));
-        ECHO(lock_release(&self->im.lock));
+            size_t nbytes = aligned_bytes_of_image(&self->im.shape);
+            if (self->kind != BasicDevice_Camera_Empty) {
+                memcpy(self->im.frame_data, self->im.render_data, nbytes);
+            }
```

Comment:
You should be able to just swap pointers here.
```suggestion
            { 
                void * const tmp = self->im.frame_data;
                self->im.frame_data = self->im.render_data;
                self->im.render_data = tmp;
            }
```